### PR TITLE
Fix/biomanu 2202 fallback to initial step on cross device

### PR DESCRIPTION
**Pull request by jpserra**
# Problem
Motion should not be enabled on mobile browsers when going for the cross device flow

# Solution
When Motion is enabled, we replace the initial face step with the `activeVideo` step. but if the user is unable to complete the IDV on desktop, we should revert motion back to the original step when users go over to mobile via cross device.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [ ] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the TESTING_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: src/components/Router/CrossDeviceMobileRouter.tsx [hunk 1]
```diff
@@ -234,10 +234,20 @@ export default class CrossDeviceMobileRouter extends Component<
       return this.setError()
     }
 
+    // NOTE, This is should be removed as soon as motion is supported on Mobile
+    const convertMotionStepToInitialConfig = (steps) => {
+      const initialFaceStep = steps.find((s) => s.type === "activeVideo")?.original;
+
+      if(!initialFaceStep)
```
Comment (warrenseine) [inline src/components/Router/CrossDeviceMobileRouter.tsx:241]: Style: run Prettier.

File: src/components/Router/CrossDeviceMobileRouter.tsx [hunk 2]
```diff
@@ -234,10 +234,20 @@ export default class CrossDeviceMobileRouter extends Component<
       return this.setError()
     }
 
+    // NOTE, This is should be removed as soon as motion is supported on Mobile
+    const convertMotionStepToInitialConfig = (steps) => {
+      const initialFaceStep = steps.find((s) => s.type === "activeVideo")?.original;
```
Comment (warrenseine) [inline src/components/Router/CrossDeviceMobileRouter.tsx:239]: How does this even compile? `StepConfig` doesn't have an `original` property.

File: src/components/Router/CrossDeviceMobileRouter.tsx (hunk 1 above)
Comment (jpserra) [reply src/components/Router/CrossDeviceMobileRouter.tsx]: fixed 👍

File: src/components/Router/CrossDeviceMobileRouter.tsx (hunk 2 above)
Comment (jpserra) [reply src/components/Router/CrossDeviceMobileRouter.tsx]: should be fixed now

**Review by warrenseine — COMMENTED** [review]
LGTM, but I'd like others to approve.

**Review by victorcamargo-onfido — APPROVED** [review]

File: src/components/Router/CrossDeviceMobileRouter.tsx [hunk 3]
```diff
@@ -234,10 +234,24 @@ export default class CrossDeviceMobileRouter extends Component<
       return this.setError()
     }
 
+    // NOTE, This is should be removed as soon as motion is supported on Mobile
```
Comment (nulrich) [inline src/components/Router/CrossDeviceMobileRouter.tsx:237]: I added `createCrossDeviceStepsHook` to group crossdevice steps "dynamic" changes.

Comment (nulrich) [inline src/components/Router/CrossDeviceMobileRouter.tsx:237]: Does your feature exists in Orchestration? If yes, how do you handle the case.

Comment (warrenseine) [reply src/components/Router/CrossDeviceMobileRouter.tsx:237]: AFAIK, no, Motion is only being A/B-tested by swapping the "face" step, no exposed through Orchestration.

**Pull request merged**